test/ssa_callback_test.jl

```diff
@@ -66,4 +66,21 @@ function paffect!(integrator)
   reset_aggregated_jumps!(integrator)
 end
 sol = solve(jprob, SSAStepper(), tstops=[1000.0], callback=DiscreteCallback(pcondit,paffect!))
-@test sol[1,end] == 100
\ No newline at end of file
+@test sol[1,end] == 100
+
+maj1 = MassActionJump([1 => 1],[1 => -1, 2 => 1]; param_idxs=1, params=p)
```

Comment:
I don't quite see why each MAJ needs a pointer to `p`.

Reply:
So it can determine the rate constant to store when being constructed (`p` isn’t actually saved internally). I guess we could store a dummy value and initialize it later, though we’d still need to know the rate’s ultimate type at the time the jump is constructed.

Reply:
So an alternative would be to optionally allow the rate field type to be passed, with a default of Float64, and then initialize the jump rate constant in the JumpProblem constructor.

Reply:
If you use `@set!` from SetField, the initialization can be to just `nothing` and it should resolve in the solve. Since it's not mutable, that should compile to be free.

Reply:
OK, I updated this so that if `p` is not passed then in the `JumpProblem` constructor `update_params!` is called to fill in the MassActionJump's rate constants from `p`.

Reply:
Would it be better to resolve in `solve` or in building the `JumpProblem` (which is what I had updated too)?

Reply:
Yeah, prob makes more sense.